### ISSUE TITLE
Support for Firefox 57

### ISSUE THREAD

**Issue by singhnsk** [Noitidart/Chrome-Store-Foxified#69]
Hi, the Addon does not seem to be working with the latest nightly 57. This is when the Web Extensions are replacing traditional addons and I was looking if I could install LastPass for Chrome to Firefox.

**Comment by tsktsktsk64** [comment]
I'm also looking forward to the support of 57.

**Comment by Noitidart** [comment]
Hi folks, I'm working on a webext version. The commits are on the main repo. I am working on multiple addons in parallel so all have been slow. :(

**Comment by tsktsktsk64** [comment]
@Noitidart I'm so happy to know about that... Great job you're doing!

**Comment by singhnsk** [comment]
@Noitidart thank you for the quick update. I hope you push it soon :)
Appreciate your work!

**Comment by chmln** [comment]
keep up the great work @Noitidart

**Comment by Noitidart** [comment]
Thanks folks! :)

**Comment by Croydon** [comment]
Duplicate of #63

**Comment by cryohellinc** [comment]
Any news? I honestly LOVE Firefox 57 so much, your addon will make Firefox my 200% daily driver, both at home and at work. :) Cant wait! Best of luck in development.

**Comment by Noitidart** [comment]
Hi @cryohellinc - its hard for me to work on this during the weekdays. You'll definitely be seeing more activity from me this weekend. :) I hope to get out a preliminary version next week, if I don't make that deadline, I'll update you :)

**Comment by cryohellinc** [comment]
@Noitidart 
Hope to see it, and good luck in development once again!

**Comment by Noitidart** [comment]
Thank you sir! :)

**Comment by chazbroam** [comment]
I know I am a complete newbie to writing Web Extension code, but how hard is it for someone like me to join in & get my hands dirty helping you write/edit/test code? I only use the Linux platform. I currently have Firefox downloaded in the following formats: Current, Beta, & Nightly. Can I help?

**Comment by Noitidart** [comment]
Thanks very much @chazbroam for asking. I made some great progress on the weekend. I don't have a test yet ready for the public. Maybe in one more week. :)

**Comment by chazbroam** [comment]
What extra things do I need to begin testing? I have Chromium, and git-2.14.1-i586 installed. Do I need Firefox-Developer Edition?

**Comment by chazbroam** [comment]
...Also started reading tutorials.

**Comment by Noitidart** [comment]
Well to test I can give you a build, but the build is not yet ready it has too many broken parts. :(

**Comment by chazbroam** [comment]
Should I start looking at the older XUL add-on to familiarize myself with the functionality? How similar/different are the 2 add-ons? Are you starting from scratch?

**Comment by Noitidart** [comment]
Thanks @chazbroam - the old XUL one is ancient, it is very different. The new one uses redux-saga, redux, react, react-router v4. It's like an app dashboard. I'll have to inlay a content script or something in order to get the same affect as the old one.

**Comment by chazbroam** [comment]
I am such a newbie, I have never heard of any of those. But, I will learn. Which one should I start learning 1st? "redux" or "react" (I'm assuming that those come 1st before redux-saga & react-router)?

In the older XUL add-on, The way it functioned was the user went to Chrome WebStore, but instead of  an "Add to Chrome" it says "Add to Firefox". I am assuming that will not change?

**Comment by Noitidart** [comment]
Oh wow that is really nice of you sir! Starting with react is a great point - http://facebook.github.io/react/

**Comment by lawrencegripper** [comment]
@Noitidart Thanks for all the hard work on this!

**Comment by Noitidart** [comment]
Thanks @lawrencegripper   I will get back to this weekend :)

**Comment by cryohellinc** [comment]
Any news? :)

**Comment by Noitidart** [comment]
Thanks for the eagerness @cryohellinc  this weekend was really busy with some family stuff so I didn't get to make any commits. :( I hate it when my plans to code get ruined :(

**Comment by cryohellinc** [comment]
@Noitidart 
Haha no worries that is fully understandable. Really looking forward to see your work continue, your addon will make Firefox 57+ best thing there is. :)

**Comment by Noitidart** [comment]
Thank you sir!! :)

**Comment by tsktsktsk64** [comment]
Feels like everyone is having an eye on you right now @Noitidart ... Again, thanks for all the efforts you put in!

**Comment by cryohellinc** [comment]
@tsktsktsk64 
When @Noitidart  goes to sleep, can he feel our breath on the back of his neck!

**Comment by Noitidart** [comment]
Thanks @tsktsktsk64 haha I love the support.

@cryohellinc hahahaha! I actually got some commits in mid week :)

Thanks all for bearing with me!

**Comment by rozariffin** [comment]
Glad to know the u working out for Firefox 57. Once the addon available for 57, there is no way i can turning back to Chrome or Opera again. I believe all of us have same thought as i am. 

Kudos and thx to dev Noitidart. 

Please keep all of us updated!

**Comment by QWERTYUIOPYOZO** [comment]
Will you be able to get this project done before the FF57 release?

**Comment by Noitidart** [comment]
Hey all,

I think I can do a really lite version. Where you put in the URL to the listing, and then it downloads it, with minimal error handling. I ran into some issues at my work so got super busy there. :(

**Comment by mehdisadeghi** [comment]
@Noitidart thanks for working on the addon! I tried to install the add-on and run it with moziall web-ext tool but I was unable to make it work. I thought I could help :/. Perhaps if you update some docs and create smaller issues people like me could help with the development.

**Comment by antiKk** [comment]
@Noitidart I'd be happy with a lite version. Thanks for the effort you're putting into this!

**Comment by svnpenn** [comment]
@Noitidart if it helps I have 2 open bounties on Super User:

- http://superuser.com/q/948192
- http://superuser.com/q/100220

they are both yours if you can solve this

**Comment by Noitidart** [comment]
Hi svnpenn I have an addon called "Trigger" which does that, but it got destroyed by the latest update. I will try to get around to make a webext version of that too. I think media keys support will be built into webext api's soon. :)

@mehdisadeghi @antiKk thank you folks so much for your patience. I am so appreciative of it. I want to deliver for you! Excuse my delay due to being swamped at work :(

**Comment by svnpenn** [comment]
@Noitidart no what I mean is that by fixing Chrome Store Foxified, it might
allow me to use the Chrome Extension StreamKeys:

http://chrome.google.com/webstore/detail/streamkeys/ekpipjofdicppbepocohdlgenahaneen

if it does, and StreamKeys works for my needs, then you will have solved my
problem, and be awarded both bounties

**Comment by Noitidart** [comment]
Ah I see, hopefully it will work, but I don't think so because it probably uses unsupported APIs in Firefox.

**Comment by dvdbot** [comment]
New guy joining the discussion,
how is the development going?

**Comment by QWERTYUIOPYOZO** [comment]
It's 12 days until FF57 release.

**Comment by dvdbot** [comment]
"Get ready. The new Firefox launches on November 14. We're excited to share it with you." - E-Mail from Firefox <Mozilla@e.mozilla.org> : "Browse into the future"

So 13 Days :P

**Comment by Noitidart** [comment]
Hey follks! Thanks @dvdbot for joining! I have an important presentation im prepping for on nov 6th, if I can't get in any work I'm going to try to make it easy to contribute for others. :(

**Comment by lawrencegripper** [comment]
@Noitidart Completely understand it's hard to balance a side project and everything else, happy to wait :) Hope the presentation goes well!

**Comment by Noitidart** [comment]
Alrighty all here is a super lite version, which doesnt support "signing". Github doesnt allow uploading .xpi so rename the attached file extension from .zip to .xpi and it will install into firefox.

I'll work on signing now. I'll also add a popup explaining that in dev edition and nightly, to get uninstalled addons to install you need to go to `about:config` and toggle pref of `xpinstall.signatures.required` to `true`. And if they are in stable or release, Ill add a dialog explaining how they can go to `about:debugging` and only install unsigned addons as "temporary" via the "load temporary addon". If anyone would like to design the popup with this explanation that would be awesome help!

[foxified-v3.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1443484/foxified-v3.zip)

**Comment by QWERTYUIOPYOZO** [comment]
I can help out with the popup, just tell me the information it needs

**Comment by Noitidart** [comment]
Thanks @QWERTYUIOPYOZO! If you can just design it that would be awesome.

the info should fit in a small dialog popup or something creative. It should tell a person "they are on stable/beta - here are the steps on how to install an unsigned addon temporarily - 1) save to disk by clicking "usnigned" then etc etc".

And a nightly popup that will say "you are on dev/nightly, you need to go to about:config and toggle pref xpinstall.signatures.required to true, if you have done that, then continue on. and then a checkbox or somethign to say "dont show dialog again"".

**Comment by QWERTYUIOPYOZO** [comment]
Can do! You are aware that content scripts cannot be injected into system pages and AMO? It will probably have to be in a separate window.

**Comment by Noitidart** [comment]
Thanks sir!

This is just a html modal ill show within the app tab. It wont user content script. :)

**Comment by Noitidart** [comment]
This i think is a complete lite version. Aside from the modal that explains how to install unsigned extensiosn on beta/stable & dev/nightly. This version makes the buttons on chrome webstore work like they did before, it reads "add to firefox":

Please test it out, I would love your feedback!

[v3 - clicking cws works.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1443820/v3.-.clicking.cws.works.zip)

**Comment by QWERTYUIOPYOZO** [comment]
Is this what you were thinking?
The Pop Up is available at - https://qwertyuiopyozo.github.io/CSFox-PopUp/
Feedback is appreciated.

**Comment by Madis0** [comment]
> https://qwertyuiopyozo.github.io/CSFox-PopUp/index.html

Some feedback:
- Grammar corrections:
  - "Welcome to Foxified Extension Stores." - no comma/period and uncapitalized "to"
  - "Install Chrome, Opera, and Edge extensions straight into Firefox." - no comma or
  - "Install Chrome, Opera, and Edge extensions, straight in Firefox." - into > in
- Animation could be a bit faster, or maybe skip some frames like the Welcome itself.
- Instead of "before you begin" warning, it would be wiser to guide the user in a tutorial slide, after which you show how to install extensions themselves.
- Image captions are not very noticable, they blend in with other text. Maybe use borders around image/caption?
- "Open the..." gets old fast. Perhaps you could just write "Chrome Web Store | Opera Addons | Microsoft Store"? Or use larger icons to indicate the stores.

**Comment by Croydon** [comment]
> As usual, github doesn't allow uploading .xpi files, so I attached it as a .zip, please download then rename to .xpi and then install to Firefox.

This is not necessary. Firefox supports installing .zip files for a while now 😄

**Comment by Noitidart** [comment]
@QWERTYUIOPYOZO hollllly moly that is beautiful!!

I actually don't need to show a dialog on welcome, but that is so cool I'll work that into a first time landing.

I need another dialog please when the user clicks on "Install Unsigned":

![](https://screenshots.firefoxusercontent.com/images/3568ade2-d94b-4233-be82-d214cab9330b.png)

In the modal it should be a guide on how to install unisnged addons based on the browser version. 

In super short summary: Nightly/dev they can flip a config. In esr/stable/beta they need go to `about:debugging` and at best can only load as "Temporary Addon".

So if the dialog can have a guide with some baby screenshoots, that would be awesome.

**Comment by Noitidart** [comment]
@QWERTYUIOPYOZO I have been watching those animations over and over again I seriously love the subtle up move on the circular arrow. So pretty!

**Comment by Noitidart** [comment]
Woohoo check this one out folks, this one supports Opera store now too! 

Edge store not yet supported :( It seems complicated.

[v3 - with ows.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1444318/v3.-.with.ows.zip)

**Comment by Croydon** [comment]
@Noitidart Have you considered renaming the add-on? I never noticed nor expected that this add-on supports anything else than the Chrome store because of it's name.

**Comment by Noitidart** [comment]
@Croydon yeah totally - https://github.com/Noitidart/Chrome-Store-Foxified/issues/71 - we're brainstorming there. I plan to release this addon to Chrome, Opera, and Edge as well. So a universal name would be awesome. I liked "Extension Stores Foxified" however that only works for Firefox. :(

**Comment by QWERTYUIOPYOZO** [comment]
Thanks so much @Noitidart! I spent a while working on it.
Thanks for the feedback @Madis0. Yes, I am conscious of the debatable grammar, however, you have a valid reason for changing it. I have updated the popup and I hope it is now closer 
It mightn't even be worth involving the Edge store, since it has basically nothing on it, most of which can be found on other stores anyway. 
I can make another popup for the unsigned installation as well, but I might not be able to get it done until tomorrow.

**Comment by QWERTYUIOPYOZO** [comment]
Oops, I just found that xpinstall.signatures.required needs to be set to false, I will update the popup again!

**Comment by Noitidart** [comment]
Thanks @QWERTYUIOPYOZO take your time you guys waited for me its only fair that i wait for you :) It may be easy work, but it takes time, i totally understand and appreciate your effort!

**Comment by QWERTYUIOPYOZO** [comment]
Thank you for your appreciation.
I'm not sure if you took a look at the source code, but the welcome dialog is semi transparent, so it will look good when added into your UI.
For future reference, what level of integration would you like my parts of the UI have with yours and should I consider this when making mine? Are your system pages just temporary for the lite version or are you going to keep them? 
Another thing I noticed when fiddling around with the latest version of the extension is that when importing Opera extensions, it has option to save to disk as chrome .crx format rather than opera .nex format, consider adding options to save to both, or otherwise just have the download option based on the extension's source.
Thanks for helping me out!
![screenshot 90](https://user-images.githubusercontent.com/29297777/32419382-1421219e-c2cd-11e7-8fd7-a809ce7c0c58.png)

**Comment by Noitidart** [comment]
I'll make a react component out of your components. So it should be one-to-one, I'll paste your html in as JSX and then CSS as CSS. :)

> Another thing I noticed when fiddling around with the latest version of the extension is that when importing Opera extensions, it has option to save to disk as chrome .crx format rather than opera .nex format, 

Oh thanks! Fixing that right now.

**Comment by QWERTYUIOPYOZO** [comment]
That sounds good, I can put the finishing touches on it now so that you can merge it in when you're ready!

**Comment by Noitidart** [comment]
Thanks!

**Comment by Noitidart** [comment]
Attached is v3 dist build, no console logging. And also it has the modals, @QWERTYUIOPYOZO check it out. The modals are empty, right now, I'll update it with your work.

[v3 - dist build and modals.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1444616/v3.-.dist.build.and.modals.zip)

**Comment by QWERTYUIOPYOZO** [comment]
> And also it has the modals, @QWERTYUIOPYOZO check it out.

You are the best @Noitidart!

Well I think I'm done with it now, so it should be fine to implement. If anybody has any last minute feedback, please let me know!

https://qwertyuiopyozo.github.io/CSFox-PopUp/index.html

**Comment by Noitidart** [comment]
Thanks this looks real cool!

But the note "Since you're using Firefox Developer Edition, you'll need to make sure that xpinstall.signatures.required is set to false in about:config before you begin." is not true, it is not required. It is only required "if they want to install unsigned". So I was thinking of putting this into the install unsigned dialog.

If beta/stable/esr then if they want to "install unsigned" they will need to download the unsigned xpi and then load it via about:debugging and "Load Temporary Addon". This I think should also go into the usngined modal.

I provided some feedback here - https://github.com/QWERTYUIOPYOZO/CSFox-PopUp/issues/1#issuecomment-342049267

Its actualy kind of important :( there is a bit of js in the welcome modal, and that needs to be in the react. I would love to help you setup the react environment so you can make this a react component! :)

**Comment by QWERTYUIOPYOZO** [comment]
It may also be good to automagically open the dashboard on first run so the user can see it.

**Comment by Noitidart** [comment]
> It may also be good to automagically open the dashboard on first run so the user can see it.

Oh thats a real important idea. I'll make sure to implement this right now.

**Comment by QWERTYUIOPYOZO** [comment]
Just fixed it, take a look!
Something else is broken though!
I'll quickly fix that

**Comment by cole-h** [comment]
@QWERTYUIOPYOZO "Install Chrome and Opera extenions, right in Firefox." -> should change "extenions" to "extensions"

**Comment by Noitidart** [comment]
Here is opening dashboard on first run:

[v3 - dist - firstrun open dashboard.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1445194/v3.-.dist.-.firstrun.open.dashboard.zip)

**Comment by cryohellinc** [comment]
Good Lord! Christmas is early this year. Thank you for all the effort guys! :)

**Comment by Noitidart** [comment]
Thank you @cryohellinc ! That's an awesome comment I really appreciate it! :) I appreicate everyones motivation and patience too! And honestly the addon needs work, it has no error handling :( So if you go offline during download it can require a disable/re-enable of the addon. :(

**Comment by QWERTYUIOPYOZO** [comment]
> @QWERTYUIOPYOZO "Install Chrome and Opera extenions, right in Firefox." -> should change "extenions" to "extensions"
Oops, fixing that now.
I also just noticed that the instruction images do not load, so I'll fix that as well

**Comment by Noitidart** [comment]
Hey @QWERTYUIOPYOZO this version here has the modal contents. I made some barebones unsigned modal content.

I made the splash pure css instead of js. However the css is a bit off. May you please use your inspector to help fix it up.

Its ready to release other then that. :)

[v3 - dist modal content.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1447431/v3.-.dist.modal.content.zip)

**Comment by Noitidart** [comment]
Actually @QWERTYUIOPYOZO i think i fixed it. Check it out - ill upload a release right now o AMO.

[v3.zip](https://github.com/Noitidart/Chrome-Store-Foxified/files/1447505/v3.zip)

**Comment by Noitidart** [comment]
Woohoo released and approved - https://addons.mozilla.org/en-US/firefox/addon/chrome-store-foxified/

This topic can finally be closed yay!

**Comment by QWERTYUIOPYOZO** [comment]
Yay!

**Comment by QWERTYUIOPYOZO** [comment]
I don't know how hard it would be to fix after you've release it, but I think the setting for first run is stored in browser.storage.sync where it may be better to have it in the local storage so that when the user re installs the addon, they can see it again (as I tried to do so I could watch my intro again).

**Comment by QWERTYUIOPYOZO** [comment]
Probably not the right place to be saying this, but anyway.

**Comment by Noitidart** [comment]
No problem man here is great to say. Changing where isFirstRun is stored is too much effort for little gain. Other important features like the community part is more important.

In Firefox uninstall is weird, after clicking uninstall, you have to close out the addon manager, then it really uninstalls. Reinstalling it will then re-trigger the weclome modal. I think disable/enable should do the same.

Also you can go to Settings > "Clear Memory" then restart browser. To see welcome again. :)

**Comment by kahing** [comment]
I am still seeing "This add-on is not compatible with your version of Firefox." and the page says the addon     was last updated "a year ago (Oct 31, 2016)"

**Comment by Noitidart** [comment]
The submission got rejected :( Submitting again now.

**Comment by Logan1x** [comment]
@Noitidart any updates on the work?
